types/block.go

```diff
@@ -78,55 +87,74 @@ func (b *Block) ValidateBasic() error {
 		)
 	}
 
-	// Validate evidence.
-	for i, ev := range b.Evidence.Evidence {
-		if err := ev.ValidateBasic(); err != nil {
-			return fmt.Errorf("Invalid evidence (#%d): %v", i, err)
-		}
+	if b.TotalTxs < b.NumTxs {
```

Comment:
that failed for me since we do not set `TotalTxs` in many tests. Maybe we should though